tests/models/test_genericconfig.py

```diff
@@ -105,38 +111,26 @@ def test_attribute_del_item(generic_config: "CustomConfig") -> None:
     assert "float" not in generic_config.configuration
 
     # "string" is defined as a pydantic model field.
-    # This means it should not be deleted, but reset to its default value
-    # and removed from the set of set fields.
+    # This means it should not be deleted from the model schema,
+    # but it should be removed from the model isntance.
     generic_config.configuration["string"] = "my secret string"
     assert generic_config.configuration.string == "my secret string"
-    assert "string" in generic_config.configuration.__fields_set__
+    assert "string" in generic_config.configuration.model_fields_set
 
```

Comment:
The behavior of deleting used to be resetting the field. And there were a lot of tests using the logic of reset. Consider adding the reset and then preserving the tests, and also adding a deprecation warning in the delete function